invenio_users_resources/administration/views/users.py

```diff
@@ -0,0 +1,152 @@
+
+    item_field_list = {
+        "spec": {"text": _("Set spec"), "order": 1},
+        "name": {"text": _("Set name"), "order": 2},
+        "search_pattern": {"text": _("Search query"), "order": 3},
+        "created": {"text": _("Created"), "order": 4},
+        "updated": {"text": _("Updated"), "order": 5},
+    }
+
+    search_config_name = "RDM_users_SEARCH"
+    search_facets_config_name = "RDM_users_FACETS"
+    search_sort_config_name = "RDM_users_SORT_OPTIONS"
+
+    create_view_name = "users_create"
+    resource_name = "name"
+
+
+class UsersEditView(AdminResourceEditView):
+    """Configuration for users sets edit view."""
+
+    name = "users_edit"
+    url = "/users/<pid_value>/edit"
+    resource_config = "users_resource"
+    pid_path = "id"
+    api_endpoint = "/users/sets"
+    title = "Edit USERS set"
+
+    list_view_name = "USERS"
+
+    form_fields = {
+        "name": {
+            "order": 1,
+            "text": _("Set name"),
+            "description": _("A short human-readable string naming the set."),
+        },
+        "spec": {
+            "order": 2,
+            "text": _("Set spec"),
+            "description": _(
+                "An identifier for the set, "
+                "which cannot be edited after the set is created."
+            ),
+        },
+        "search_pattern": {
+            "order": 3,
+            "text": _("Search query"),
+            "description": _(
+                "See the supported query "
+                "syntax in the "
+                "<a href='/help/search'>Search Guide</a>."
+            ),
+        },
+        "created": {"order": 4},
+        "updated": {"order": 5},
+    }
+
+
+class UsersCreateView(AdminResourceCreateView):
+    """Configuration for users sets create view."""
+
+    name = "users_create"
+    url = "/users/create"
+    resource_config = "users_resource"
+    pid_path = "id"
+    api_endpoint = "/users/sets"
+    title = "Create USERS set"
+
+    list_view_name = "USERS"
+
+    form_fields = {
+        "name": {
+            "order": 1,
+            "text": _("Set name"),
+            "description": _("A short human-readable string naming the set."),
+        },
+        "spec": {
+            "order": 2,
+            "text": _("Set spec"),
+            "description": _(
+                "An identifier for the set, "
+                "which cannot be edited after the set is created."
+            ),
+        },
+        "search_pattern": {
+            "order": 3,
+            "text": _("Search query"),
+            "description": _(
+                "See the supported query "
+                "syntax in the <a href='/help/search'>Search Guide</a>."
+            ),
+        },
+    }
+
+
+class UsersDetailView(AdminResourceDetailView):
+    """Configuration for users sets detail view."""
+
+    url = "/users/<pid_value>"
+    api_endpoint = "/users/sets"
+    search_request_headers = {"Accept": "application/json"}
+    name = "USERS details"
```

Comment:
general comment on `name` if it is not a user facing attribute it shoudl be lowercased and snakecased (i.e. no spaces, but `_`). It applies to all views' name
```suggestion
    name = "User details"
```

Reply:
Done